File: sdk-typescript/test/iwfcompat/state_options_test.ts
// Copyright (c) 2026 Super Durable, Inc.
//
// Licensed under the Super Durable Source License 1.0.
// You may not use this file except in compliance with the License.
// See the LICENSE file in the repository root.
//
// SPDX-License-Identifier: LicenseRef-Super-Durable-1.0

import { stringCodec, type Client } from "../../src/index.js";

import * as flows from "./iwf_flows.js";

export async function compileTimeoutRetryDurabilityAndLocks(
  client: Client,
): Promise<void> {
  await client.startFlow(flows.STATE_OPTIONS, "state-options", undefined);
  const output: string = await client.waitForFlow("state-options", stringCodec);
  void output;
}

File: sdk-java/src/test/java/io/superdurable/dex/iwfcompat/StateOptionsTest.java
/*
 * Copyright (c) 2026 Super Durable, Inc.
 *
 * Licensed under the Super Durable Source License 1.0.
 * You may not use this file except in compliance with the License.
 * See the LICENSE file in the repository root.
 *
 * SPDX-License-Identifier: LicenseRef-Super-Durable-1.0
 */

package io.superdurable.dex.iwfcompat;

import io.superdurable.dex.Client;

public final class StateOptionsTest {
    void compileTimeoutRetryDurabilityAndLocks(final Client client) {
        client.startFlow(IwfFlows.STATE_OPTIONS, "state-options", null);
        final String output = client.waitForFlow("state-options", String.class);
        consume(output);
    }

    private static void consume(final Object value) {
    }
}
